### PR TITLE
AB2D-1202 code-climate improvement

### PR DESCRIPTION
**Pull request by spathiyil**
Create a parameter object for passing parameters from `ContractProcessorImpl` to `PatientClaimsProcessorImpl`

### Summary

Create a parameter object class named PatientClaimsRequest
that encapsulates all the parameters that need to be passed
from ContractProcessorImpl to PatientClaimsProcessorImpl

Adjust the junit tests to use the new request parameter

### Checklist

- [x] Tests and linting pass

### Security
N/A

### Additional JIRA Tickets (optional)

N/A

### CODE REVIEW

File: worker/src/main/java/gov/cms/ab2d/worker/processor/PatientClaimsProcessorImpl.java [hunk 1]
```diff
@@ -41,50 +41,50 @@
     private final BFDClient bfdClient;
     private final FhirContext fhirContext;
 
-    @Trace(async = true)
-    @Async("patientProcessorThreadPool")
     /**
      * Process the retrieval of patient explanation of benefit objects and write them
      * to a file using the writer
      */
-    public Future<Void> process(PatientDTO patientDTO, final StreamHelper helper, OffsetDateTime attTime,
```
Comment (adaykin) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/PatientClaimsProcessorImpl.java]: Is the purpose of this to just reduce the # of method parameters?

Comment (spathiyil) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/PatientClaimsProcessorImpl.java]: yes

File: worker/src/main/java/gov/cms/ab2d/worker/processor/domainmodel/PatientClaimsRequest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package gov.cms.ab2d.worker.processor.domainmodel;
+
+import com.newrelic.api.agent.Token;
+import gov.cms.ab2d.worker.adapter.bluebutton.GetPatientsByContractResponse;
+import gov.cms.ab2d.worker.processor.StreamHelper;
+
+import java.time.OffsetDateTime;
+
+public class PatientClaimsRequest {
```
Comment (dkrylovsb) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/domainmodel/PatientClaimsRequest.java]: Can we use Lombok to auto-generate the constructor and the getters?

Comment (spathiyil) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/domainmodel/PatientClaimsRequest.java]: done

**Pull request merged**